marimbabot_vision/scripts/flatten_whiteboard_data.py

```diff
@@ -1,25 +1,33 @@
+import argparse
 import os
-import tqdm
-from multiprocessing import Pool
 import re
+from multiprocessing import Pool
 from shutil import copyfile
 
-NUM_WORKER = 30
-INPUT_DIR = 'data'
-OUTPUT_DIR = 'data_wb'
-NAME_PREFIX = 'staff_1'
+import tqdm
+
+parser = argparse.ArgumentParser(
+                    prog = 'flatten_whiteboard_data.py',
+                    description = 'This script generates a separate dataset including one sample folder for each captured whiteboard picture. Each sample folder contains the correspoding .txt file, the .ly file and the .png file.')
+
+parser.add_argument('-o', '--outdir', type=str, required=True, help='The output directory for storing the whiteboard dataset')
+parser.add_argument('-i', '--indir', type=str, required=True, help='The input directory containing the captured whiteoard images')
+parser.add_argument('-p', '--prefix', type=str, default='staff_1', help='The prefix of the file names. Default: staff_1')
+parser.add_argument('-w', '--workers', type=int, default=30, help='The number of workers. Default: 30')
 
 def main(sample):
-    sample_folder = os.path.join(INPUT_DIR, sample)
+
+    sample_folder = os.path.join(args.indir, sample)
     for real_im in [i for i in os.listdir(sample_folder) if os.path.isfile(os.path.join(sample_folder,i)) and 'real' in i]:
-        num = re.search(f'{NAME_PREFIX}_real_(.*).png', real_im).group(1)
-        real_sample_folder = os.path.join(OUTPUT_DIR, f'{sample}_{num}')
+        num = re.search(f'{args.prefix}_real_(.*).png', real_im).group(1)
+        real_sample_folder = os.path.join(args.outdir, f'{sample}_{num}')
         os.makedirs(real_sample_folder, exist_ok=True)
-        copyfile(os.path.join(sample_folder, f'{NAME_PREFIX}.txt'), os.path.join(real_sample_folder, f'{NAME_PREFIX}.txt'))
-        copyfile(os.path.join(sample_folder, f'{NAME_PREFIX}.ly'), os.path.join(real_sample_folder, f'{NAME_PREFIX}.ly'))
-        copyfile(os.path.join(sample_folder, real_im), os.path.join(real_sample_folder, f'{NAME_PREFIX}.png'))
-        # os.remove(os.path.join(sample_folder, real_im)) # !!! only after whiteboard data collection is finished !!!
+        copyfile(os.path.join(sample_folder, f'{args.prefix}.txt'), os.path.join(real_sample_folder, f'{args.prefix}.txt'))
+        copyfile(os.path.join(sample_folder, f'{args.prefix}.ly'), os.path.join(real_sample_folder, f'{args.prefix}.ly'))
+        copyfile(os.path.join(sample_folder, real_im), os.path.join(real_sample_folder, f'{args.prefix}.png'))
 
 if __name__ == "__main__":
-    with Pool(NUM_WORKER) as pool:
-        list(tqdm.tqdm(pool.imap(main, os.listdir(INPUT_DIR)), total=len(os.listdir(INPUT_DIR))))
+    args = parser.parse_args()
```

Comment:
Did you check that everything works with this scoping? Because `args` is defined down here and not explicitly passed to the functions.

Also consider moving the parser definition into the if name == main thing.

Reply:
I tested it and it worked, but I was wondering about that too.
I put the parser definition on top as some sort of substitute for a script description, but I guess this is a kind of inelegant solution.

Reply:
I moved the parser stuff to the main function now. Tested once more, still works.